src/service/resources-impl.js

```diff
@@ -1495,8 +1476,7 @@ export class Resource {
     /** @const {!AmpElement|undefined|null} */
     this.owner_ = undefined;
 
-    /** @const {number} */
-    this.priority_ = getElementPriority(element.tagName);
+    this.priority_ = element.getElementPriority();
```

Comment:
Important!!! This, unfortunately, will not work. The priority here is initialized at the construction time and thus there's no guarantee that the element has been upgraded. So, what's likely to happen is that, e.g. an `amp-analytics` element will initially be initialized with `StubElement` instance which will return the value of `0` which will be stored here. After upgrade, the instance will be switched to `AmpAnalyticsElement`, but it's priority will no longer be queried. This is most likely why I had at first implemented this in a method above. A bit of a challenge here is that the task system currently in place is agnostic to the type of task. In other words, the system assumes that it could be used for more than just layout, and possibly before upgrade happened. In reality, so far only layout uses this task system. So, we can just specialize around layout when element has been already upgraded.

So, to do this, we should:

1. We can keep method name as `getPriority()`, but we need to document clearly that it's only available when element has been upgraded.
2. We need to override `getPriority()` method in the `StubElement` class and throw an exception there, e.g. "Can only be called on an upgraded element". Alternatively, we can do the same in the `ElementProto` in `custom-element.js` - I believe there are bunch of asserts like this already. This is critical to ensure we avoid regressions.
3. `Resource` class can no longer initialize `this.priority_` in the constructor. Instead, it should simply delegate to the `ElementProto -> BaseElement` when asked. If element is not upgraded it will simply fail and we will fix whatever might get broken.